### ISSUE TITLE
TestRegress170 fails under latest bridge changes

### ISSUE THREAD

**Issue by VenelinMartinov** [pulumi/pulumi-hcloud#528]
### Describe what happened

https://github.com/pulumi/pulumi-hcloud/actions/runs/9451894942/job/26034396438?pr=525

Seems to expect no diff but gets a diff.

Looks like a regression on https://github.com/pulumi/pulumi-hcloud/issues/170

### Sample program

.

### Log output

```
  "response": {
    "stables": "*",
    "changes": "DIFF_SOME",
    "hasDetailedDiff": true,
    "diffs": ["networks"],
    "detailedDiff": {
      "networks[0].networkId": {
          "kind": "UPDATE"
    }
  }
  },
```

### Output of `pulumi about`

.

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by VenelinMartinov** [comment]
Reproes with bridge master

**Comment by iwahbe** [comment]
P1 because it's blocking the next bridge release.

**Comment by iwahbe** [comment]
I have confirmed that the regression is genuine (and not overfitting from the test).

**Comment by iwahbe** [comment]
A binary search has shown that "`ba48631b` Enable DiffStrategy PlanState by default ([#1976](https://github.com/pulumi/pulumi-terraform-bridge/pull/1976))" is the last good commit on the bridge, "`b012fa8e` Fix proposed new unknown blocks ([#2060](https://github.com/pulumi/pulumi-terraform-bridge/pull/2060))" introduced the regression.

**Comment by iwahbe** [comment]
This looks like a bug in upstream. I have reproduced using TF.
